packages/widget/src/pages/ErrorPage/ErrorPageBadPriceWarning.tsx

```diff
@@ -49,6 +60,53 @@ export const ErrorPageBadPriceWarning = ({
     tokenAmount: amountOut,
   });
 
+  const { title, descriptionContent } = useMemo(() => {
+    if (hasUsdValues && swapDifferencePercentage) {
+      return {
+        title: `Warning: Bad trade (-${swapDifferencePercentage})`,
+        descriptionContent: (
+          <>
+            You will lose {swapDifferencePercentage} of your input value with this trade
+            <br />
+            Input: {sourceDetails?.amount} {sourceDetails?.symbol} ({usdAmountIn})
+            <br />
+            Estimated output: {destinationDetails?.amount} {destinationDetails?.symbol} ({usdAmountOut})
+          </>
+        ),
+      };
+    } else if (priceImpactPercentage) {
+      return {
+        title: `Warning: High Price Impact (${priceImpactPercentage})`,
+        descriptionContent: (
+          <>
+            Executing this trade is expected to impact the price by {priceImpactPercentage}. Please verify the amounts.
+            <br />
+
+          </>
+        ),
+      };
+    } else {
+      return {
+        title: "Warning: Bad Trade",
+        descriptionContent: (
+          <>
+            This trade may result in a poor execution price. Please verify the amounts carefully.
+            <br />
+          
+          </>
+        ),
+      };
+    }
```

Comment:
nit: this is purely a stylistic thing, but in the cases where each if condition has an early return I prefer this:
```ts
if () {
  return 1;
}
if () {
  return 2;
}
return 3;
```

over 

```ts
if () {
  return 1;
} else if () {
  return 2;
} else {
  return 3;
}

Reply:
oh you updated it before I finished writing my comment 😆